primitives/core/src/bandersnatch.rs

```diff
@@ -0,0 +1,1016 @@
+		let s = self.to_ss58check();
+		write!(f, "{} ({}...)", crate::hexdisplay::HexDisplay::from(&self.as_ref()), &s[0..8])
+	}
+
+	#[cfg(not(feature = "std"))]
+	fn fmt(&self, _: &mut sp_std::fmt::Formatter) -> sp_std::fmt::Result {
+		Ok(())
+	}
+}
+
+/// Bandersnatch signature.
+///
+/// The signature is created via the [`VrfSecret::vrf_sign`] using [`SIGNING_CTX`] as `label`.
+#[cfg_attr(feature = "full_crypto", derive(Hash))]
+#[derive(Clone, Copy, PartialEq, Eq, Encode, Decode, PassByInner, MaxEncodedLen, TypeInfo)]
+pub struct Signature([u8; SIGNATURE_SERIALIZED_LEN]);
+
+impl UncheckedFrom<[u8; SIGNATURE_SERIALIZED_LEN]> for Signature {
+	fn unchecked_from(raw: [u8; SIGNATURE_SERIALIZED_LEN]) -> Self {
+		Signature(raw)
+	}
+}
+
+impl AsRef<[u8]> for Signature {
+	fn as_ref(&self) -> &[u8] {
+		&self.0[..]
+	}
+}
+
+impl AsMut<[u8]> for Signature {
+	fn as_mut(&mut self) -> &mut [u8] {
+		&mut self.0[..]
+	}
+}
+
+impl TryFrom<&[u8]> for Signature {
+	type Error = ();
+
+	fn try_from(data: &[u8]) -> Result<Self, Self::Error> {
+		if data.len() != SIGNATURE_SERIALIZED_LEN {
+			return Err(())
+		}
+		let mut r = [0u8; SIGNATURE_SERIALIZED_LEN];
+		r.copy_from_slice(data);
+		Ok(Self::unchecked_from(r))
+	}
+}
+
+impl ByteArray for Signature {
+	const LEN: usize = SIGNATURE_SERIALIZED_LEN;
+}
+
+impl CryptoType for Signature {
+	#[cfg(feature = "full_crypto")]
+	type Pair = Pair;
+}
+
+impl sp_std::fmt::Debug for Signature {
+	#[cfg(feature = "std")]
+	fn fmt(&self, f: &mut sp_std::fmt::Formatter) -> sp_std::fmt::Result {
+		write!(f, "{}", crate::hexdisplay::HexDisplay::from(&self.0))
+	}
+
+	#[cfg(not(feature = "std"))]
+	fn fmt(&self, _: &mut sp_std::fmt::Formatter) -> sp_std::fmt::Result {
+		Ok(())
+	}
+}
+
+/// The raw secret seed, which can be used to reconstruct the secret [`Pair`].
+#[cfg(feature = "full_crypto")]
+type Seed = [u8; SEED_SERIALIZED_LEN];
+
+/// Bandersnatch secret key.
+#[cfg(feature = "full_crypto")]
+#[derive(Clone)]
+pub struct Pair {
+	secret: SecretKey,
+	seed: Seed,
+}
+
+#[cfg(feature = "full_crypto")]
+impl Pair {
+	/// Get the key seed.
+	pub fn seed(&self) -> Seed {
+		self.seed
+	}
+}
+
+#[cfg(feature = "full_crypto")]
+impl TraitPair for Pair {
+	type Seed = Seed;
+	type Public = Public;
+	type Signature = Signature;
+
+	/// Make a new key pair from secret seed material.
+	///
+	/// The slice must be 64 bytes long or it will return an error.
+	fn from_seed_slice(seed_slice: &[u8]) -> Result<Pair, SecretStringError> {
+		if seed_slice.len() != SEED_SERIALIZED_LEN {
+			return Err(SecretStringError::InvalidSeedLength)
+		}
+		let mut seed = [0; SEED_SERIALIZED_LEN];
+		seed.copy_from_slice(seed_slice);
+		let secret = SecretKey::from_seed(&seed);
+		Ok(Pair { secret, seed })
+	}
+
+	/// Derive a child key from a series of given (hard) junctions.
+	///
+	/// Soft junctions are not supported.
+	fn derive<Iter: Iterator<Item = DeriveJunction>>(
+		&self,
+		path: Iter,
+		_seed: Option<Seed>,
```

Comment:
Maybe a silly question, but why is this argument ignored? Perhaps put a comment here and/or return an error if it is `Some`?

Reply:
Tools which uses the `InspectKeyCmd` may end up calling (after some hopping) the [from_string_with_seed](https://github.com/paritytech/substrate/blob/ece92d6b30ee7f9fbda7a84bee76f8ff6d2c92a1/primitives/core/src/crypto.rs#L953) which in turn will call `pair.derive(junctions, seed)`.

With:
- `seed` the seed generated from the secret phrase according to bip39.
- Junctions are the hard/soft junctions after the phrase

Is left to the implementation what to do with the seed during the `derive` which has been already used to generate the pair itself. 

The only implementation which uses it is `sr25519` and frankly I don't know why it requires it.
But I don't find it of any usage for bandersnatch as to derive a subkey we use the junctions.

Returning an error is not an option since is called in the same way for all the possible pairs